examples/step-81/doc/results.dox

```diff
@@ -0,0 +1,298 @@
+the real and the imaginary components. We started off using the $H^{curl}$
+conforming Nédélec Elements, and we made two copies of the Finite Elements
+in order to represent the real and the imaginary components of our input
+(FE_NedelecSZ was used instead of FE_Nedelec to avoid the sign conflicts
+issues present in traditional Nédélec elements). In the assembly, we create
+two vectors of dimension $dim$ that assist us in extracting the real and
+the imaginary components of our finite elements.
+
+
+<h4> Rotations and Scaling </h4>
+As we see in our assembly, our finite element is rotated and scaled as
+follows:
+
+@code
+const auto phi_i = real_part.value(i, q_point) - 1.0i * imag_part.value(i, q_point);
+@endcode
+
+This $\phi_i$ variable doesn't need to be scaled in this way, we may choose
+any arbitrary scaling constants $a$ and $b$. If we choose this scaling, the
+$\phi_j$ must also be modified with the same scaling, as follows:
+
+@code
+const auto phi_i = a*real_part.value(i, q_point) -
+                                   bi * imag_part.value(i, q_point);
+
+const auto phi_j = a*real_part.value(i, q_point) +
+                                   bi * imag_part.value(i, q_point);
+@endcode
+
+Moreover, the cell_rhs need not be the real part of the rhs_value. Say if
+we modify to take the imaginary part of the computed rhs_value, we must
+also modify the cell_matrix accordingly to take the imaginary part of temp.
+However, making these changes to both sides of the equation will not affect
+our solution, and we will still be able to generate the surface plasmon
+polariton.
+
+@code
+cell_rhs(i) += rhs_value.imag();
+
+cell_matrix(i) += temp.imag();
+@endcode
+
+<h4> Postprocessing </h4>
+We will create a video demonstrating the wave in motion, which is
+essentially an implementation of $e^{-i\omega t}(Re(E) + i*Im(E))$ as we
+increment time. This is done by slightly changing the output function to
+generate a series of .vtk files, which will represent out solution wave as
+we increment time. Introduce an input variable $t$ in the output_results()
+class as output_results(unsigned int t). Then change the class itself to
+the following:
+
+@code
+template <int dim>
+void Maxwell<dim>::output_results(unsigned int t)
+{
+  std::cout << "Running step:" << t << std::endl;
+  DataOut<2> data_out;
+  data_out.attach_dof_handler(dof_handler);
+  Vector<double> postprocessed;
+  postprocessed.reinit(solution);
+  for (unsigned int i = 0; i < dof_handler.n_dofs(); ++i)
+    {
+      if (i % 4 == 0)
+        {
+          postprocessed[i] = std::cos(2 * M_PI * 0.04 * t) * solution[i] -
+                             std::sin(2 * M_PI * 0.04 * t) * solution[i + 1];
+        }
+      else if (i % 4 == 2)
+        {
+          postprocessed[i] = std::cos(2 * M_PI * 0.04 * t) * solution[i] -
+                             std::sin(2 * M_PI * 0.04 * t) * solution[i + 1];
+        }
+    }
```

Comment:
This is not particular nice. The result vector is empty every second entry and the function makes an implicit assumption how the components are sorted. What about showing a fool-proof implementation based on the FiniteElement functions. One might want to enumerate the real and imaginary blocks on their own to be able to use a block preconditioner.